elastic/kibana
# [Infra UI] Show only hosts with metrics collected by system module in the hosts view

Closes #176403

## Summary

This PR adds a filter for the `event.module` to be `system` because the Hosts View is only compatible with the metrics-system indices - I added a [comment](https://github.com/elastic/kibana/issues/176403#issuecomment-1954232722) to explain the change in the query. It adds infra client as part of the synthtrace and a scenario to test the change

## Testing 
- Use the new synthtrace scenario: `node scripts/synthtrace --clean infra_hosts_with_apm_hosts.ts`
   - By default there should be `10` host visible on the host view and 3 separate services in APM (the APM hosts should not be visible) 
   - The scenario can be used with different numbers of services/hosts for example: 
`node scripts/synthtrace --clean --scenarioOpts.numServices=5 --scenarioOpts.numHosts=5 infra_hosts_with_apm_hosts.ts` 
       - 5 hosts shown on Infrastructure >  Hosts  (the APM hosts should not be visible) 
       ![image](https://github.com/elastic/kibana/assets/14139027/d8763a24-95c2-43cd-991b-23edd102f47a)
       - 5 services shown on APM > Services
       ![image](https://github.com/elastic/kibana/assets/14139027/0638cfdb-3d6f-4f72-915d-d67764bc9349)
- Use remote cluster (with APM)
   - The hosts with `0` metrics coming from APM should not be visible:
   <img width="1920" alt="image" src="https://github.com/elastic/kibana/assets/14139027/af69efc0-bbd9-47ae-8431-2a56fa0626c4">
